pkg/api/error.go

```diff
@@ -43,6 +46,15 @@ type CloudErrorBody struct {
 	Details []CloudErrorBody `json:"details,omitempty"`
 }
 
+type ValidationStatus string
```

Comment:
Ok so I may have mislead you partially previously.  

So ARM dictates we must respond with a [specific error response content](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content).  This doesn't include the newly added `status` field on the CloudError struct.  

So my suggestion is to create a new type, but nest the CloudError struct within it and add the field `status` necessary within it. 

This shouldn't generate new swagger specifications because that would require an API change.  

The thing that confuses me is ARM expects all error responses as [ODatav4.0 json format response](http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091), but for this specific one it expects the typical error response plus an additional field.  I would have hoped it would be common.  

nesting the struct within a new struct should solve our problem and introducing the additional `status` field should be sufficient here though.  

Apologies for the misleading previously.